source/more_sampling_tools.Rmd

```diff
@@ -209,13 +211,513 @@ message(n_golds / (n_golds + n_silvers))
 :::
 
 
+## Back to Boolean {{< var array >}}s
+
+The code above implements the procedure we might well use if were were
+simulating the problem physically.  We do a trial, and we record the result.
+We do this on a piece of paper if we are doing a physical simulation, and in
+the `second_trunks` {{< var array >}} in code.
+
+Finally we tally up the results.  If we are doing a physical simulation, we go
+back over the all the trial results and counting up the "Gold" and "Silver"
+outcomes.  In code we use the comparisons `== 'Gold'` and `== 'Silver'` to find
+the trials of interest, and then count them up with {{< var sum >}}.
+
+Boolean {{< var array >}}s are a fundamental tool in {{< var lang >}}, and we will use them in nearly all our simulations.
+
+Here is a remind of how those {{< var array >}}s work.
+
+First, let's slice out the first 10 values of the `second_trunks`
+trial-by-trial results tally from the simulation above:
+
+```{python}
+# Get values at positions 0 through 9 (up to, but not including position 10)
+first_10_trunks = second_trunks[:10]
+first_10_trunks
+```
+
+```{r}
+# Get values at positions 1 through 10
+first_10_trunks <- second_trunks[1:10]
+first_10_trunks
+```
+
+Before we started the simulation, we set `second_trunks` to contain 10,000
+strings, where each string was "No gold in first trunk".  In the simulation, we
+check whether there was gold in the first trunk, and, if not, we don't change
+the value in `second_trunk`, and the value remains as "No gold in first trunk".
+
+Only if there was gold in the first trunk, do we go on to check whether the
+second trunk contains silver or gold.  Therefore, we only set a new value in `second_trunks` where there was gold in the first trunk.
+
+Now let us show the effect of running a comparison on `first_10_trunks`:
```

Comment:
let us -> let's

Reply:
Thanks!  I think it's 50-50 here - partly because Simon has a rather old-fashioned style I'm trying to reproduce in part - but happy to go for your half of the 50-50.

```suggestion
Now let's show the effect of running a comparison on `first_10_trunks`:
```